test/tests/functional/pbs_strict_ordering.py

```diff
@@ -204,6 +206,42 @@ def test_zero_backfill_depth_on_queue(self):
         self.server.expect(JOB, {'comment': job_comment}, id=jid3, offset=2,
                            max_attempts=2, interval=2)
 
+        # Now try the same scenario with backfilling set to one and check
+        # that first and third job runs but second gets calendared.
+        # since we want thrid job to backfill around second, we need to make
+        # sure that walltime of third job is less than the walltime of first
+        # job
+        a = {'scheduling': 'False'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+        self.server.deljob([jid1, jid2, jid3])
+
+        a = {'backfill_depth': 1}
+        self.server.manager(MGR_CMD_SET, QUEUE, a, id="workq")
+
+        a = {'Resource_List.select': '1:ncpus=1',
+             'Resource_List.walltime': '100'}
+        j4 = Job(TEST_USER)
+        j4.set_attributes(a)
+        jid4 = self.server.submit(j4)
+
+        a = {'Resource_List.select': '1:ncpus=2'}
+        j5 = Job(TEST_USER)
+        j5.set_attributes(a)
+        jid5 = self.server.submit(j5)
+
+        a = {'Resource_List.select': '1:ncpus=1',
+             'Resource_List.walltime': '50'}
+        j6 = Job(TEST_USER)
+        j6.set_attributes(a)
+        jid6 = self.server.submit(j6)
+
+        a = {'scheduling': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid4)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid6)
+        self.scheduler.log_match(jid5 + ";Job is a top job")
```

Comment:
If you submit your these 3 jobs as your first 3 jobs (they're the same but with walltime), you don't need these.  You can reuse the same 3 jobs for both sides of this test.

Reply:
I feel the way it is right now is cleaner because in the other implementation I'll have to make sure the jobs have enough sleep/walltime so that on slower machines they don't end in between and for log match I'll have to make sure that we specify the timestamp.
I'd like to retain it as is because functionality wise I feel it is not different than keeping the same jobs across cycles.

Reply:
OK, I'll buy that.  Thanks for explaining.